presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowFlightClientHandler.java

```diff
@@ -0,0 +1,202 @@
+                if (config.getArrowFlightServerSslEnabled() != null && !config.getArrowFlightServerSslEnabled()) {
+                    location = Location.forGrpcInsecure(config.getFlightServerName(), config.getArrowFlightPort());
+                }
+                else {
+                    location = Location.forGrpcTls(config.getFlightServerName(), config.getArrowFlightPort());
+                }
+            }
+
+            logger.debug("location %s", location.getUri().toString());
+
+            FlightClient.Builder flightClientBuilder = FlightClient.builder(allocator, location);
+            if (config.getVerifyServer() != null && !config.getVerifyServer()) {
+                flightClientBuilder.verifyServer(false);
+            }
+            else if (config.getFlightServerSSLCertificate() != null) {
+                trustedCertificate = Optional.of(new FileInputStream(config.getFlightServerSSLCertificate()));
+                flightClientBuilder.trustedCertificates(trustedCertificate.get()).useTls();
+            }
+
+            FlightClient flightClient = flightClientBuilder.build();
+            this.arrowFlightClient = new ArrowFlightClient(flightClient, trustedCertificate);
+            isClientClosed.set(false);
+        }
+        catch (Exception ex) {
+            throw new ArrowException(ARROW_FLIGHT_ERROR, "The flight client could not be obtained." + ex.getMessage(), ex);
+        }
+    }
+
+    public ArrowFlightConfig getConfig()
+    {
+        return config;
+    }
+
+    public ArrowFlightClient getClient(Optional<String> uri)
+    {
+        if (isClientClosed.get()) { // Check if client is closed or not initialized
+            logger.info("Reinitialize the client if closed or not initialized");
+            initializeClient(uri);
+            scheduleCloseTask();
+        }
+        else {
+            resetTimer(); // Reset timer when client is reused
+        }
+        return this.arrowFlightClient;
+    }
+
+    public FlightInfo getFlightInfo(ArrowFlightRequest request, ConnectorSession connectorSession, Optional<String> bearerToken)
+    {
+        try {
+            ArrowFlightClient client = getClient(Optional.empty());
+            CredentialCallOption auth = this.getCallOptions(connectorSession, bearerToken);
+            FlightDescriptor descriptor = FlightDescriptor.command(request.getCommand());
+            logger.debug("Fetching flight info");
+            FlightInfo flightInfo = client.getFlightClient().getInfo(descriptor, ArrowFlightConstants.CALL_OPTIONS_TIMEOUT, auth);
+            logger.debug("got flight info");
+            return flightInfo;
+        }
+        catch (Exception e) {
+            throw new ArrowException(ARROW_FLIGHT_ERROR, "The flight information could not be obtained from the flight server." + e.getMessage(), e);
+        }
+    }
+
+    public CredentialCallOption getCallOptions(ConnectorSession connectorSession, Optional<String> bearerToken) throws IOException
```

Comment:
Let's make this method abstract, so that an implementing class is responsible for setting authentication properties and returning call options.